rack_hack/src/com/thoughtworks/go/rackhack/DelegatingServlet.java

```diff
@@ -39,10 +38,43 @@ public void init(ServletConfig config) throws ServletException {
         rackServlet.init(config);
     }
 
+    private static Method method(String name, Class klass) throws NoSuchFieldException {
+        if (klass == null) {
+            return null;
+        }
+        Method[] fields = klass.getDeclaredMethods();
+        for (Method field : fields) {
+            if (field.getName().equals(name)) {
+                field.setAccessible(true);
+                return field;
+            }
+        }
+        return method(name, klass.getSuperclass());
+    }
+
+    public static Object invoke(Object o, String method, Object... args) throws Exception {
```

Comment:
Since we know that this invoke is called only from this class (right?), we can remove the variable-length args and just let it be String. I'm thinking about the performance of this, since this is called every request. Reflection is reasonably quick these days, but still, maybe there's no need to make this method so generic, without reason?